cloudscape-design/components
# feat: adding variant colors to badge

### Description

This pull request extends the Badge component in the "cloudscape-design/components" repository by adding new color variants, including `critical`, `high`, `medium`, `low`, and `neutral`. These new variants provide users with more options to customize the appearance of the Badge component according to different severity levels or states.

### How has this been tested?

The changes have been tested using the following steps:

1. Set up a development environment for the "cloudscape-design/components" repository.
2. Create a test scenario for each new color variant (`critical`, `high`, `medium`, `low`, `neutral`).
3. Integrate the Badge component with different elements and use the new color variants.
4. Verify that the Badge component renders correctly with the expected colors for each variant.
5. Ensure that the changes do not impact the functionality or appearance of the existing Badge component.

### Screenshots
| Design Spec | Running demo assets |
|--|--|
| <img width="627" alt="Screen Shot 2023-06-19 at 11 33 27" src="https://github.com/cloudscape-design/components/assets/75797/cdebd0bf-1d8b-422d-8bb0-bc661bdf90f8">| <img width="287" alt="Screen Shot 2023-06-19 at 11 43 14" src="https://github.com/cloudscape-design/components/assets/75797/75391b57-ebd6-442c-b8a8-b4d49846f695">|


#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
